packages/vc-verification/src/resolver/issuer-resolver.ts

```diff
@@ -40,15 +48,28 @@ export class EthersProviderIssuerResolver implements IssuerResolver {
   async getIssuerDefinition(
     namespace: string
   ): Promise<IIssuerDefinition | undefined> {
+    const cachedRoleDefinition = this._dataCache?.getRoleDefinition(namespace);
+    if (cachedRoleDefinition) {
+      return cachedRoleDefinition.issuer;
+    }
     const resolvedNamespace = namespace.startsWith('0x')
       ? namespace
       : utils.namehash(namespace);
     const roleDefinition = await this._domainReader.read({
       node: resolvedNamespace,
     });
     if (DomainReader.isRoleDefinitionV2(roleDefinition)) {
+      this._dataCache?.setRoleDefinition(namespace, roleDefinition);
       return roleDefinition.issuer;
     }
     return undefined;
   }
+
+  /**
+   * Sets intermediate cache
+   * @param dataCache
+   */
+  setCache(dataCache: IDataCache): void {
```

Comment:
I think this is an Interface Segregation Principle violation because `EthersProviderIssuerResolver` only needs the `RoleDefinition` cache but it is exposed to the `DIDDocument` and `RoleCredential` cache methods.
I think a way to solve this would be as follows (given interface segregation proposed in my previous comment):
```suggestion
  setRoleDefinitionCache(roleDefinitionCache: IRoleDefinitionCache): void {
```